controllers/databasecluster_controller.go

```diff
@@ -341,6 +341,15 @@ func (r *DatabaseClusterReconciler) Reconcile(ctx context.Context, req ctrl.Requ
 	}
 	logger.Info("Reconciled", "request", req)
 	_, ok := database.ObjectMeta.Annotations[restartAnnotationKey]
+	database.ObjectMeta.Labels = map[string]string{
+		databaseClusterNameLabel: database.Name,
+	}
+	if database.Spec.DataSource != nil {
+		database.ObjectMeta.Labels[fmt.Sprintf(backupStorageNameLabelTmpl, database.Spec.DataSource.ObjectStorageName)] = "used"
+	}
+	for _, schedule := range database.Spec.Backup.Schedules {
+		database.ObjectMeta.Labels[fmt.Sprintf(backupStorageNameLabelTmpl, schedule.ObjectStorageName)] = "used"
+	}
```

Comment:
Should we clean up labels when the ObjectStorageName is not used anymore?

Reply:
We should. Great catch. Thanks